### DISCUSSION TITLE
What is the default "connection-timeout" value ?

### DISCUSSION THREAD

**Discussion by sub-bucket** [spring-projects/spring-amqp#2424] [Q&A]
Hello,
  I'm a programmer from china , i try to study rabbimtmq ,i use springboot ,but when i didn't set the key "spring.rabbitmq.connection-timeout", i can't connect to the rabbitmq server, so i set the key to 30000ms, it works.
and i read the  sources ,i didn't find where is the default value for "spring.rabbitmq.connection-timeout", can you tell me where is
the default value  is seted.
thanks

**Comment by artembilan** [comment]
Spring Boot auto-configuration delegates to the `RabbitConnectionFactoryBean`:

```
	map.from(this.rabbitProperties::getConnectionTimeout).whenNonNull().asInt(Duration::toMillis)
				.to(factory::setConnectionTimeout);
```

That one, in turn, to the `com.rabbitmq.client.ConnectionFactory`:
```
	/**
	 * @param connectionTimeout the connection timeout.
	 * @see com.rabbitmq.client.ConnectionFactory#setConnectionTimeout(int)
	 */
	public void setConnectionTimeout(int connectionTimeout) {
		this.connectionFactory.setConnectionTimeout(connectionTimeout);
	}
```
And this one has a default like this:
```
    /** The default TCP connection timeout: 60 seconds */
    public static final int    DEFAULT_CONNECTION_TIMEOUT = 60000;
```

Not sure what difference it makes for you, but probably a connection problem is not related to this timeout value.

Please, consider next time to ask questions on StackOverflow.

**Reply by sub-bucket** [reply]
> Spring Boot auto-configuration delegates to the `RabbitConnectionFactoryBean`:
> 
> ```
> 	map.from(this.rabbitProperties::getConnectionTimeout).whenNonNull().asInt(Duration::toMillis)
> 				.to(factory::setConnectionTimeout);
> ```
> 
> That one, in turn, to the `com.rabbitmq.client.ConnectionFactory`:
> 
> ```
> 	/**
> 	 * @param connectionTimeout the connection timeout.
> 	 * @see com.rabbitmq.client.ConnectionFactory#setConnectionTimeout(int)
> 	 */
> 	public void setConnectionTimeout(int connectionTimeout) {
> 		this.connectionFactory.setConnectionTimeout(connectionTimeout);
> 	}
> ```
> 
> And this one has a default like this:
> 
> ```
>     /** The default TCP connection timeout: 60 seconds */
>     public static final int    DEFAULT_CONNECTION_TIMEOUT = 60000;
> ```
> 
> Not sure what difference it makes for you, but probably a connection problem is not related to this timeout value.
> 
> Please, consider next time to ask questions on StackOverflow.

Thanks so much !
